src/process.rs

```diff
@@ -39,21 +39,62 @@ impl Drop for TemporaryProcess {
     }
 }
 
-pub struct LaunchOptions<'a> {
+pub struct LaunchOptions {
     pub headless: bool,
     pub port: Option<u16>,
-    pub path: &'a str,
+    pub path: std::path::PathBuf,
 }
 
-impl<'a> Default for LaunchOptions<'a> {
-    fn default() -> Self {
-        LaunchOptions {
+impl LaunchOptions {
+    pub fn default_executable() -> Option<std::path::PathBuf> {
+        // TODO BSDs/Unixes are the same?
+        #[cfg(target_os = "linux")]
+        {
+            // TODO Look at $BROWSER and if it points to a chrome binary
+            // $BROWSER may also provide default arguments, which we may
+            // or may not override later on.
+
+            // TODO More paths!?
+            let default_paths = &["/usr/bin/google-chrome-stable", "/usr/bin/chromium-browser"][..];
+            for path in default_paths {
+                if std::path::Path::new(path).exists() {
+                    return Some(path.into());
+                }
+            }
+        }
+
+        #[cfg(target_os = "macos")]
+        {
+            let default_paths =
+                &["/Applications/Google Chrome.app/Contents/MacOS/Google Chrome"][..];
+            for path in default_paths {
+                if std::path::Path::new(path).exists() {
+                    return Some(path.into());
+                }
+            }
+        }
+        // TODO Windows
+        // Check HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\App Paths\chrome.exe
+        None
+    }
+
+    pub fn default() -> Option<Self> {
+        Some(LaunchOptions {
             headless: true,
             // TODO: extra option for if you want it to keep scanning up from the port you passed?
             port: None,
-            // TODO: this is not at all a sensible default
-            path: "/Applications/Google Chrome.app/Contents/MacOS/Google Chrome",
-        }
+            path: Self::default_executable()?,
+        })
+    }
+
+    pub fn headless(mut self, headless: bool) -> Self {
+        self.headless = headless;
+        self
+    }
+
+    pub fn port(mut self, port: Option<u16>) -> Self {
+        self.port = port;
+        self
     }
```

Comment:
```suggestion
    }
    
    pub fn path(mut self, path: std::path::PathBuf) -> Self {
        self.path = path;
        self
    }
```

For when users want everything default except for the path (e.g. sometimes I switch between versions of Chrome).